src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -976,4 +976,89 @@
       it { expect(response).to redirect_to(project_show_path(apache_maintenance_incident_project)) }
     end
   end
+
+  describe 'POST #move_path' do
+    context 'valid project' do
+      let(:repository) { create(:repository, project: apache_project) }
+      let(:path_element) { create(:path_element, repository: repository) }
+
+      context 'without direction' do
+        it 'raises an exception' do
+          expect { post :move_path, params: { project: apache_project } }.to raise_error ActionController::ParameterMissing
+        end
+      end
+
+      context 'only one path_element' do
+        let(:position) { path_element.position}
+
+        context 'direction up' do
+          before do
+            post :move_path, params: { project: apache_project, repository: repository.id, direction: 'up', path: path_element }
+          end
+
+          it 'goes up' do
+            expect(path_element.reload.position).to eq(position)
```

Comment:
I would make this three different tests, then it is easier to debug in the future:

```
it { expect(path_element.reload.position).to eq(position) }
it { ... }
```